PWGHF/D2H/Tasks/taskDstarToD0Pi.cxx

```diff
@@ -253,15 +280,45 @@ struct HfTaskDstarToD0Pi {
     // MC at Generator Level
     for (const auto& mcParticle : rowsMcPartilces) {
       if (TESTBIT(std::abs(mcParticle.flagMcMatchGen()), aod::hf_cand_dstar::DecayType::DstarToD0Pi)) { // MC Matching is successful at Generator Level
+
         auto ptGen = mcParticle.pt();
         // auto yGen = mcParticle.y(); // Can we use this definition?
         auto yGen = RecoDecay::y(mcParticle.pVector(), o2::constants::physics::MassDStar);
         if (yCandDstarGenMax >= 0. && std::abs(yGen) > yCandDstarGenMax) {
           continue;
         }
+
+        auto mcCollision = mcParticle.mcCollision_as<aod::McCollisions>();
+        auto recCollisions = collisions.sliceBy(colsPerMcCollision, mcCollision.globalIndex());
+        // auto recCollisions = collisions.sliceByCached(aod::mccollisionlabel::mcCollisionId, mcCollision.globalIndex(), cache);
+        // auto recCollisions = collisions.sliceByCachedUnsorted(aod::mccollisionlabel::mcCollisionId, mcCollision.globalIndex(), cache);
+
+        // looking if a generated collision reconstructed more than a times.
+        if (recCollisions.size() > 1) {
+          for (const auto& [c1, c2] : combinations(CombinationsStrictlyUpperIndexPolicy(recCollisions, recCollisions))) {
+            auto deltaCent = abs(c1.centFT0M() - c2.centFT0M());
+            registry.fill(HIST("QA/hDeltaCentGen"), deltaCent);
+          }
+        }
+
+        float centFT0MGen;
+        if (recCollisions.size()) {
+          // assigning centrality to MC Collision using max FT0M amplitute from Reconstructed collisions
+          std::vector<float> colCentVec;
+          for (const auto& recCol : recCollisions) {
+            colCentVec.push_back(recCol.centFT0M());
+          }
+          std::sort(colCentVec.begin(), colCentVec.end());
+          centFT0MGen = colCentVec[0]; // Selecting most central collision amplitute
```

Comment:
Here you should select the collision with the larger number of PV contributors (`collision.numContrib()`) and not the larger centrality

Reply:
I tried to make the changes that were needed. Please look at this if the method is correct.

Reply:
It looks good to me, thanks!